### DISCUSSION TITLE
Telemetry in rider plugin

### DISCUSSION THREAD

**Discussion by Socolin** [reqnroll/Reqnroll#39] [Contributor Q&A] [answered]
Hello,

Tricentis added some telemetry thing using Application Insights on the rider plugin to know how many user there were etc... I was never happy about it but I did not have much to say, I would like to know if we should remove this or if there is a need for such telemetry.

If we keep it I think it should changes to an opt-in option.

Any though on that ?

The code is here
https://github.com/reqnroll/Reqnroll.Rider/tree/main/src/dotnet/ReSharperPlugin.ReqnrollRiderPlugin/Analytics

**Comment by gasparnagy** [comment, marked as answer]
Telemetry is always a tricky question. I think if the result is used well it can be a super useful tool to make decisions or detect issues. (E.g. without that it would be impossible to tell, how much percent of our users use .NET Framework 4 still, but it helped a lot already to see typical errors the users receive.) This is telemetry that helps us to know if the thing we build operates at the users as we expected. We don't track users.

For this reason I left the telemetry in for the Reqnroll Core and the Reqnroll VS extension, but I created a new telemetry key so the data is not going to the Tricentis service anymore. My plan is to share the stats regularly with the community.

As you are the main contributor of this codebase, I leave it up to you if you want to disable this. And you can change your decision later as well. 

If you choose to keep it, please make sure that you replace the Tricentis dev key to ours, that you can find [here](https://github.com/reqnroll/Reqnroll/blob/main/Reqnroll/Reqnroll.csproj#L62). This is only the "dev" key, used only for testing the telemetry. During release this is replaced with the "prod" key that is stored in a GitHub secret. We can setup access to the data for you as well.

But again, I am happy with both options.

**Comment by gasparnagy** [comment]
@Socolin Did you left the telemetry in finally or removed it?

**Reply by Socolin** [reply]
For now I kept the code (by default: no effort), I don't know if it's configure on a AppInsights or not but I'm not using the data anyway. If nobody have a use for it I might delete it.

**Comment by Code-Grump** [comment]
Where is the telemetry key configured?

**Comment by gasparnagy** [comment]
I did check it and it was disabled in the build and also it was using a wrong dev key (local builds do not report to the prod telemetry, but there is a special "dev" playground). I have fixed it now.

But in general, this is the way to go. In the reqnroll GitHub org there is a secret registered with our production AppInsights key: `${{ secrets.APPINSIGHTS_KEY }}` and that should be provided to the build (like in the commented line).

I would love to include the stats into my "monthly stats" (like https://github.com/orgs/reqnroll/discussions/121), so let's see if it is providing valuable results after I re-enabled it now.
